pkg/validation/admission.go

```diff
@@ -0,0 +1,72 @@
+package validation
+
+import (
+	"context"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	metal3v1alpha1 "github.com/metal3-io/baremetal-operator/pkg/apis/metal3/v1alpha1"
+)
+
+// ValidateHostForAdmission returns an error if the host fails uniqueness
+// validation, and cannot be added to the cluster.  This is ordinarily run by an
+// admission webhook.
+func ValidateHostForAdmission(c client.Client, host metal3v1alpha1.BareMetalHost) error {
+	if host.Spec.BMC.Address == "" && host.Spec.BootMACAddress == "" && host.Spec.BMC.CredentialsName == "" {
+		return nil
+	}
+
+	ctx := context.Background()
+	hostList := metal3v1alpha1.BareMetalHostList{}
+	opts := &client.ListOptions{
+		Namespace: host.Namespace,
```

Comment:
We need to search all namespaces here because nothing about the Host being in another namespace makes it better when they're both fighting for control over Ironic. In fact, that's probably worse.